### PR TITLE
explicit --data flag instead of `~/.tendermint` (Refs #431)

### PR DESCRIPTION
**Pull request by melekes**

### CODE REVIEW

File: cmd/tendermint/commands/run_node.go [hunk 1]
```diff
@@ -35,27 +35,27 @@ var (
 func init() {
 
 	// configuration options
-	runNodeCmd.Flags().StringVar(&moniker, "moniker", config.GetString("moniker"),
```
Comment (rigelrozanski) [inline cmd/tendermint/commands/run_node.go]: What's the logic surrounding changing all of the default flag values from dependent on the config to hard-coded?

Comment (melekes) [inline cmd/tendermint/commands/run_node.go]: Given that path to config is now dynamic (`--data path`; `path/config.toml`), I don't see a way to read it beforehand and display default values cause we don't know the path to config, doh.

Maybe we could create an instance of default Config in memory and get defaults from it.  How does that sound? Or maybe there is a way to solve this puzzle using Viper?

Comment (rigelrozanski) [inline cmd/tendermint/commands/run_node.go]: Config is a global variable within `commands`, so long as it has been initialized we should be able to retrieve values from it without knowing what path it was derived from. We can make sure that it is set beforehand by doing all the data flag processing in the `cobra.command.prerun` I believe.

Comment (rigelrozanski) [inline cmd/tendermint/commands/run_node.go]: Oops. Yeah I get why this a problem now. The only option I see where the defaults can be initialized from config is if we define a persistent config file location (which could be modified with a tendermint command). This way the config file would be initialized before the flags and could be read in... maybe not as nice as just using the `--data` flag. As for initializing from a default config, this seems unwieldy - the current hard coded defaults seems reasonable for the time being

Comment (melekes) [inline cmd/tendermint/commands/run_node.go]: Ok, thanks

**Comment by ebuchman** [comment]
Closing as this will be handled by new viper work.

**Pull request closed without merging**